hybrasyl/Objects/ConditionInfo.cs

```diff
@@ -0,0 +1,191 @@
+﻿using Hybrasyl.Enums;
+using Hybrasyl.Statuses;
+using Newtonsoft.Json;
+
+namespace Hybrasyl.Objects
+{
+    [JsonObject(MemberSerialization.OptIn)]
+    public class ConditionInfo
+    {
+        public Creature Creature { get; set; }
+        public User User => Creature as User;
+
+        [JsonProperty]
+        public CreatureCondition Conditions { get; set; }
+
+        [JsonProperty]
+        public PlayerFlags Flags { get; set; }
+
+        private void _initialize()
+        {
+            Conditions = 0;
+            Flags = PlayerFlags.Alive;
+        }
+        public ConditionInfo(Creature owner)
+        {
+            Creature = owner;
+            _initialize();
+        }
+
+        public ConditionInfo(User user)
+        {
+            Creature = User as Creature;
+            _initialize();
+        }
+
+        public bool CastingAllowed
+        {
+            get
+            {
+                var conditionCheck = Asleep || Frozen || Paralyzed || Comatose;
+
+                if (User != null)
+                    conditionCheck = conditionCheck || ((Flags & PlayerFlags.ProhibitCast) != 0);
+                return conditionCheck;
+            }
+        }
+
+        public bool IsAttackable
+        {
+            get
+            {
+                if (User != null)
+                    return PvpEnabled;
```

Comment:
This should also check for whether or not AscendEnabled.

AscendEnabled likely needs to be created.

Reply:
Why does ascending need to be any different than having pvp enabled?

Reply:
It doesn't need to be. I figure though it might be something better to have say if someone is PvpEnabled and also wants to Ascend. Having PvpEnabled should function differently than ascension in my opinion.

Example: PvpEnabled death should not allow the user to access Ascension related areas.